libsolidity/codegen/ExpressionCompiler.cpp

```diff
@@ -1580,6 +1580,20 @@ bool ExpressionCompiler::visit(MemberAccess const& _memberAccess)
 			m_context << Instruction::DUP1 << u256(32) << Instruction::ADD;
 			utils().storeStringData(contract.name());
 		}
+		else if (member == "interfaceId")
+		{
+			TypePointer arg = dynamic_cast<MagicType const&>(*_memberAccess.expression().annotation().type).typeArgument();
+			ContractDefinition const& contract = dynamic_cast<ContractType const&>(*arg).contractDefinition();
+			u256 result{0};
+			for (auto& function : contract.definedFunctions()) {
```

Comment:
Please use the external interface instead.

Reply:
interfaceFuncitonList, this also provides you with the hash already.

Reply:
Ah yes, I used this at first, but then I saw that derived functions where also included. However, I changed now `ContractDefinition::interfaceFunctionList()`.